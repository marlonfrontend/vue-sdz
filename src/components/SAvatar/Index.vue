<template>
  <div :class="classes">
    <slot />
  </div>
</template>

<script>
export default {
  name: 'SAvatar',

  props: {
    size: {
      type: String,
      validator: function (value) {
        return ['p', 'm', 'g'].indexOf(value) !== -1
      },
      default: 'm'
    },

    color: {
      type: String,
      validator: function (value) {
        return ['p', 'm', 'g'].indexOf(value) !== -1
      },
      default: 'm'
    }

    // value: {
    //   type: String,
    //   validator: function (value) {
    //     return ['p', 'm', 'g'].indexOf(value) !== -1
    //   }
    // }
  },

  computed:{
    classes(){
      return [
        's-avatar',
        `--is-size-${this.size}`,
        `--is-color-${this.color}`
      ]
    }
  }

}
</script>

<style lang="scss">
@import "./src/styles/_index.scss";

.s-avatar {
  border-radius: 100px;
  display: flex;
  background: color(primary, base);
  height: 40px;
  width: 40px;

  & > .icon{
    color: color(base, neutral)
  }

  & .dark{
    background: color(base, neutral);
  }
}
</style>
