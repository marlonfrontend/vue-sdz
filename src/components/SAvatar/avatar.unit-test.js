import { mount } from '@vue/test-utils'

import SAvatar from './Index.vue'

/* eslint-disable */
describe('SAvatar', () => {

  let cmp;

  beforeEach(() => {
    cmp = mount(SAvatar, {
      propsData: {
        size: 'm',
        color: 'dark'
      }
    });
  });


  test('is instantiated', () => {
    expect(cmp.exists()).toBeTruthy()
  })
  

  test("has the expected html structure", () => {
    expect(cmp.element).toMatchSnapshot();
  })

  test('check size m', () => {
    expect(cmp.classes('--is-size-p')).toBe(false)
    expect(cmp.classes('--is-size-g')).toBe(false)
    expect(cmp.classes('--is-size-m')).toBe(true)
    expect(cmp.classes()).toContain('--is-size-m')
  })

  test('check color dark', () => {
    expect(cmp.classes('--is-color-light')).toBe(false)
    expect(cmp.classes('--is-color-dark')).toBe(true)
    expect(cmp.classes()).toContain('--is-color-dark')
  })

})
