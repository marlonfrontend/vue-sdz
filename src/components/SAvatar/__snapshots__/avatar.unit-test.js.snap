// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SAvatar has the expected html structure 1`] = `
<div
  class="s-avatar --is-size-m --is-color-dark"
/>
`;
